src/app/shared/services/iam.service.ts

```diff
@@ -116,58 +121,48 @@ export class IamService {
 
           retVal = true;
         }
-      }
-      catch (e) {
+      } catch (e) {
         console.error(e);
-        this.logout();
-        location.reload();
+        this.toastr.error('Could not login, please try again later');
       }
-    }
-    else {
+    } else {
       // The account address is set so it means the user is current loggedin
       retVal = true;
     }
 
     return retVal;
   }
 
+  private get isUserPresent(): boolean {
+    return Boolean(this._user.getValue());
+  }
+
   async setupUser() {
-    // No need to setup user again
-    if (this._didDocument) {
+    if (this.isUserSettedUp) {
       return;
     }
 
-    // Setup DID Document
     this._didDocument = await this._iam.getDidDocument();
-    this.store.dispatch(userClaims.setDidDocument({didDocument: this._didDocument}));
-    // Get User Claims
-    let data: any[] = await this.iam.getUserClaims();
-    // console.log('getUserClaims()', JSON.parse(JSON.stringify(data)));
-
-    // Get Profile Related Claims
-    data = data.filter((item: any) => item.profile ? true : false );
-    // console.log('Profile Claims', JSON.parse(JSON.stringify(data)));
-
-    // Get the most recent claim
-    if (data.length) {
-        let tmp: any = data[0].profile;
-        this._user.next({
-          name: tmp.name,
-          birthdate: new Date(tmp.birthdate),
-          address: tmp.address
-      });
-    }
-    else {
-      this._user.next(undefined);
-    }
+    this.store.dispatch(userClaimsActions.setDidDocument({didDocument: this._didDocument}));
+    this.store.dispatch(userClaimsActions.loadUserClaims());
+  }
+
+  private get isUserSettedUp(): boolean {
```

Comment:
Slightly better grammar
```suggestion
  private get isUserSetUp(): boolean {
```